### ISSUE TITLE
Additional 'case' blocks cannot appear after the 'default' block of a 'switch'

### ISSUE THREAD

**Issue by dpasirst** [apple/swift-openapi-generator#211]
It's possible the upstream tool is the root cause as it's using Aide in Rust creating OpenAPI 3.1 that is then being downgraded to OpenAPI 3.0 - but other generators don't seem to have a problem with this.  
The problem is that when the openapi code has `default` specified before something like a `200` response, the generated code creates the `default` in the Swift switch case before the other options which is not valid in the Swift language. I'm not sure if having default first is correct or not in the OpenAPI spec, but it does not seem to be a problem with other generators, spec validators, and tools like apimatic.io - its also automatically generated this way so manually maintaining a different version is less than ideal.

In this case, the generated OpenAPI spec has the following:

```yaml
paths:
  /api/info:
    get:
      summary: version
      description: server version information
      operationId: version
      parameters: []
      responses:
        default:
          description: ''
          headers: {}
          content:
            application/json:
              schema: {}
        '200':
          description: plain text
          headers: {}
          content:
            text/plain; charset=utf-8:
              schema: {}
        '401':
          description: not authorized if secured
          headers: {}
          content: {}
      deprecated: false
```

The result when the swift open-api-generator runs is:
`Additional 'case' blocks cannot appear after the 'default' block of a 'switch'`

as the generated code looks like this:
```swift
    /// version
    ///
    /// server version information
    ///
    /// - Remark: HTTP `GET /api/info`.
    /// - Remark: Generated from `#/paths//api/info/get(version)`.
    public func version(_ input: Operations.version.Input) async throws -> Operations.version.Output
    {
        try await client.send(
            input: input,
            forOperation: Operations.version.id,
            serializer: { input in
                let path = try converter.renderedRequestPath(template: "/api/info", parameters: [])
                var request: OpenAPIRuntime.Request = .init(path: path, method: .get)
                suppressMutabilityWarning(&request)
                try converter.setHeaderFieldAsText(
                    in: &request.headerFields,
                    name: "accept",
                    value: "application/json, text/plain"
                )
                return request
            },
            deserializer: { response in
                switch response.statusCode {
                default:
                    let headers: Operations.version.Output.Default.Headers = .init()
                    let contentType = converter.extractContentTypeIfPresent(
                        in: response.headerFields
                    )
                    let body: Operations.version.Output.Default.Body
                    if try contentType == nil
                        || converter.isMatchingContentType(
                            received: contentType,
                            expectedRaw: "application/json"
                        )
                    {
                        body = try converter.getResponseBodyAsJSON(
                            OpenAPIRuntime.OpenAPIValueContainer.self,
                            from: response.body,
                            transforming: { value in .json(value) }
                        )
                    } else {
                        throw converter.makeUnexpectedContentTypeError(contentType: contentType)
                    }
                    return .`default`(
                        statusCode: response.statusCode,
                        .init(headers: headers, body: body)
                    )
                case 200:
                    let headers: Operations.version.Output.Ok.Headers = .init()
                    let contentType = converter.extractContentTypeIfPresent(
                        in: response.headerFields
                    )
                    let body: Operations.version.Output.Ok.Body
                    if try contentType == nil
                        || converter.isMatchingContentType(
                            received: contentType,
                            expectedRaw: "text/plain"
                        )
                    {
                        body = try converter.getResponseBodyAsText(
                            Swift.String.self,
                            from: response.body,
                            transforming: { value in .text(value) }
                        )
                    } else {
                        throw converter.makeUnexpectedContentTypeError(contentType: contentType)
                    }
                    return .ok(.init(headers: headers, body: body))
                case 401:
                    let headers: Operations.version.Output.Unauthorized.Headers = .init()
                    return .unauthorized(.init(headers: headers, body: nil))
                }
            }
        )
    }
```

**Comment by czechboy0** [comment]
Yes, I'd consider this a bug in Swift OpenAPI Generator. We should always move the default case to the end of the array before we generate the cases.

Probably should be solved by adding a new computed property `responseOutcomes` on `OperationDescription` (like we already have for parameters), and there we should move the default case, if present, to the very end. Otherwise the order should be preserved.

Then in places in the client and server translator where we get responses for a given operation, we start using the new property on operation description.

A PR would be welcome!

**Comment by dpasirst** [comment]
Perhaps there should be a little more thought to this. `default` response from an api server seems like something different than a `default` case in a switch statement which is more like a catch-all. With that in mind, I keep wondering if it would be better to somehow map `default` to `defaultServerResponse` or something similar. Thus, the `.undocumented` case could also work as expected.

**Comment by czechboy0** [comment]
This distinction already exists, between `default` and ``default``.

Here we really just need to change the order and it'll fix this bug.

The undocumented case is only generated when no default responses was specified in the OpenAPI doc.

**Comment by czechboy0** [comment]
Addressed by @dpasirst in #212 - thanks! 🙏
